wasm_runtime/src/c_api.rs

```diff
@@ -388,19 +397,33 @@ pub extern "C" fn wasm_executionctx_stdin_set(executionctx_id: *const c_char, bu
 /// wasm_return_const_char_ownership(module_output);
 /// ```
 #[no_mangle]
-pub extern "C" fn wasm_executionctx_run(executionctx_id: *const c_char) -> *const c_char {
+pub extern "C" fn wasm_executionctx_run(executionctx_id: *const c_char, _buffer: &mut *const c_char, _len: &mut c_ulong) -> c_int {
     let executionctx_id_str = const_c_char_to_str(executionctx_id);
 
-    let result = match WasmExecutionCtx::run(executionctx_id_str) {
-        Ok(output) => output,
+    let result: c_int = match WasmExecutionCtx::run(executionctx_id_str) {
+        Ok(output) => {
+            // 1) extract output length and try casting into `c_ulong` since it might not fit
+            *_len = match c_ulong::try_from(output.len()) {
+                Ok(l) => l,
+                Err(e) => {
+                    let error_msg = format!("ERROR! C-API: Couldn't cast Wasm module's output length ({}) into a C `unsigned long int` type! {:?}", output.len(), e);
+                    eprintln!("{}", error_msg); 
+                    return -1; // we end function execution here and ignore the output buffer
+                }
+            };
+            
+            // 2) extract output
+            *_buffer = vec_u8_to_const_c_char(output);
+            0
```

Comment:
I only have one question regarding this mechanism: `from_vec_unchecked` -- used by `vec_u8_to_const_c_char` documents to append a `\0` at the end. The len set in this function, should be `output.len()`, or `output.len() + 1`?

Reply:
Good question! Indeed, that was my first implementation but it's wrong. It must be `output.len()` since the `_len` we return must behave as `strlen()`, and thus, not counting the trailing NULL terminator.

If we add +1, that would mean '\0' is part of the string returned by the module, and at a high level, 
HTTP requests won't work, since they will find a '\x00' byte not expected. That convention only applies to the C arrays.